### PR TITLE
[RW-8239][risk=no] Consolidated DAR and AAR pages

### PR DESCRIPTION
**Pull request by evrii**
Description:
Removed AAR page and redistributed all necessary contents.

Initial Registration:
image
image

Annual Renewal
image
image
image
image

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 1]
```diff
@@ -375,18 +385,27 @@ export const getEligibleModules = (
...
+        !isRenewalCompleteForModule(
+          getAccessModuleStatusByNameOrEmpty(
+            profile.accessModules.modules,
+            moduleName
+          )
+        ))
   );
 
 // exported for test
 export const getActiveModule = (
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:404]: we probably don't want to replicate this behavior (only one module can be the active one) for renewal ... let's think about whether this is OK as an iteration.

File: ui/src/app/pages/access/access-renewal.spec.tsx [hunk 2]
```diff
@@ -1,544 +0,0 @@
-import * as React from 'react';
```
Comment (jmthibault79) [inline ui/src/app/pages/access/access-renewal.spec.tsx:1]: these tests have important logic checks - how can we ensure that tests for the combined page cover these cases?

File: ui/src/app/pages/access/modules-for-annual-renewal.tsx [hunk 3]
```diff
@@ -1,18 +1,482 @@
 import * as React from 'react';
```
Comment (jmthibault79) [inline ui/src/app/pages/access/modules-for-annual-renewal.tsx:1]: is this all copy-paste from annual-renewal?

**Review by jmthibault79 — COMMENTED** [review]
Some initial thoughts

File: ui/src/app/pages/access/access-renewal.spec.tsx (hunk 2 above)
Comment (evrii) [reply ui/src/app/pages/access/access-renewal.spec.tsx:1]: Good call. I will verify that these tests are covered elsewhere.

File: ui/src/app/pages/access/modules-for-annual-renewal.tsx (hunk 3 above)
Comment (evrii) [reply ui/src/app/pages/access/modules-for-annual-renewal.tsx:1]: Yeah, this is copy and paste.

File: ui/src/app/pages/access/access-renewal.spec.tsx [hunk 4]
```diff
@@ -1,544 +0,0 @@
...
-
-    // complianceTraining is not shown because it is disabled
-    expect(findNodesByExactText(wrapper, 'Complete Training').length).toBe(0);
-
-    // all of the necessary steps = 3 rather than the usual 4
-    expectComplete(wrapper);
-    expectNotExpired(wrapper);
-  });
-
-  // RW-7473: sync expiring/expired Training module to gain access
```
Comment (jmthibault79) [inline ui/src/app/pages/access/access-renewal.spec.tsx:496]: I don't see a corresponding test for these cases in the new file

**Review by jmthibault79 — COMMENTED** [review]
Nice mov demo!

Behavior looks correct for these test cases when running locally:
- User who is complete and CT-ineligible
- User who is complete (except for CT Training) and CT-eligible (Bypass All appears, and CT Training is available)
- Expired and Expiring modules look correct
- Multiple expiring/expiring modules do not direct the user to choose one in particular in renewal mode, as intended
- Clicking Renewal "Get Started" when on the *initial* mode page correctly switches to renewal mode.
- Signing thee DUCC in renewal mode does return the user to renewal mode

A few small concerns:
- the Access Renewal notification "Get Started" button should be disabled on the renewal page (I bet it's trying to match the old path)
- merging main brought in an empty profile/data-user-code-of-conduct.tsx file which should be removed (b/c I moved the file recently)
- please restore the RW-7473 tests
- the (existing) logic to determine which modules are "active", "required", "clickable", and "eligible" do not find Renewal Mode well.  I would not be surprised if there was a subtle edge case bug hiding in there.  We should revisit this soon. 
- the green "you have completed" box needs a little more space from the top header, in renewal mode.

initial mode:
initial-top

renewal mode:
renewal-top

**Comment by evrii** [comment]
image
I updated the header of the Annual renewal page to match the initial page. I figured this would address the spacing that you had mention, but it would also help it to look more consistent. What do you think?

**Comment by jmthibault79** [comment]
new header LGTM

File: ui/src/app/pages/signed-in/access-renewal-notification.tsx [hunk 5]
```diff
@@ -17,14 +17,19 @@ export const AccessRenewalNotificationMaybe = () => {
         ? daysRemaining + ' days remaining.'
         : 'Your access has expired.'
     }`;
+
+  // Must use pathname and search because ACCESS_RENEWAL_PATH includes a path with a search parameter.
+  const { pathname, search } = window.location;
+  const fullPagePath = pathname + search;
```
Comment (jmthibault79) [inline ui/src/app/pages/signed-in/access-renewal-notification.tsx:23]: nice

File: ui/src/app/pages/access/modules-for-annual-renewal.tsx [hunk 6]
```diff
@@ -25,13 +546,21 @@ export const ModulesForAnnualRenewal = (props: RenewalCardProps) => {
   };
   return (
     <FlexColumn style={styles.modulesContainer}>
-      {modules.map((moduleName) => {
+      {modules.map((moduleName, index) => {
         // TODO RW-7797.  Until then, hardcode
         const showModule = true;
         return (
           <FlexColumn
```
Comment (jmthibault79) [inline ui/src/app/pages/access/modules-for-annual-renewal.tsx:553]: existing problem, but when we have an array of components, we should supply a unique `key` prop to each to ensure that React renders them correctly.  Here I think it would be as simple as adding `key={index}` to this FlexColumn.  I expect `moduleName` would also work.

More info if you are not familiar: https://reactjs.org/docs/lists-and-keys.html

File: ui/src/app/pages/access/modules-for-annual-renewal.tsx [hunk 7]
```diff
@@ -1,18 +1,539 @@
...
+    columnGap: '0.25rem',
+    gridTemplateColumns: 'auto 1rem 1fr',
+    alignItems: 'center',
+  },
+  complianceTrainingExpiring: {
+    borderTop: `1px solid ${colorWithWhiteness(colors.dark, 0.8)}`,
+    marginTop: '0.5rem',
+    paddingTop: '0.5rem',
+  },
+  hippo: {
```
Comment (jmthibault79) [inline ui/src/app/pages/access/modules-for-annual-renewal.tsx:135]: hippo?

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 8]
```diff
@@ -175,17 +175,17 @@ export const styles = reactStyles({
     paddingRight: '0.5em',
   },
   clickableModuleBox: {
-    padding: '0.5em',
-    margin: '0.2em',
+    padding: '1.4em',
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:178]: This changes the Initial-Registration mode padding as well.  Is that desired?

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 9]
```diff
@@ -204,6 +204,10 @@ export const styles = reactStyles({
     opacity: '0.5',
     fontSize: '12px',
   },
+  moduleStatus: {
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:207]: this is for renewal specifically, so how about renewalStatus?

File: ui/src/app/pages/access/data-access-requirements.tsx [hunk 10]
```diff
@@ -497,17 +528,78 @@ export const DataDetail = (props: { icon: string; text: string }) => {
 
 export const DataAccessRequirements = fp.flow(withProfileErrorModal)(
   (spinnerProps: WithSpinnerOverlayProps) => {
+    // Local State
+    // At any given time, at most two modules will be clickable:
```
Comment (jmthibault79) [inline ui/src/app/pages/access/data-access-requirements.tsx:532]: shouold clarify that this does not apply to renewal

**Review by jmthibault79 — APPROVED** [review]
Looks great!
